lib/tasks/user.rake

```diff
@@ -1,12 +1,21 @@
 # frozen_string_literal: true
 
 namespace :user do
-  desc 'Updates the first_admin_id attribute value for all providers.'
+  desc 'Updates the first_admin_id attribute value for all accounts.'
   task update_all_first_admin_id: :environment do
-    Account.transaction do
-      Account.providers.find_each do |account|
-        account.update!(first_admin_id: account.first_admin&.id)
-      end
-    end
+    impersonation_admin_username = ThreeScale.config.impersonation_admin['username']
+    complete_query = <<~SQL
```

Comment:
Why not just removing the transaction earlier and logging the failed accounts.
I think that should be safe enough. This update will probably do a table lock